### ISSUE TITLE
Do not override existing `pyproject.toml` keys during conversion

### ISSUE THREAD

**Issue by rmartin16** [beeware/briefcase#1898]
### What is the problem or limitation you are having?

When an existing project is converted, keys should not be added to `[tool.briefcase]` when they already exist in `[project]`. For instance, if `version` is already specified in `[project]`, do not add `version = "0.0.1"` to `[tool.briefcase]`. This should be considered for all PEP-621 `pyproject.toml` keys.

### Describe the solution you'd like

Omit `[tool.briefcase]` keys from `pyproject.toml` that are already represented in `[project]`.

### Describe alternatives you've considered

Manually remove overriding keys from `[tool.briefcase]` post-conversion.

**Comment by freakboy3742** [comment]
Conceptually a good idea; but could be difficult to implement; the mechanism we're using to append `pyproject.toml` is (AFAIR) concatenation, rather than parsing and processing, so removing duplicate keys requires some pre-processing or a formatting-sensitive parser.

It may be better to consider this part of a broader migration to PEP621 format, so that a `version` field in `[tool.briefcase]` is no longer required or preferred.

**Comment by CarliJoy** [comment]
I think this is solved already by https://github.com/beeware/briefcase/pull/2772

**Comment by freakboy3742** [comment]
@CarliJoy Not really. This issue is about the `convert` command, which takes an existing project with a `pyproject.toml` configuration and adds a Briefcase configuration. In the process, it duplicates some of the keys (like `version`) which *can* be picked up from the `[project]` section, but can't easily be removed from the templated `[tool.briefcase]` section that the `convert` command adds to the existing configuration.

**Comment by CarliJoy** [comment]
🤦🏻  Sorry my bad.
